### DISCUSSION TITLE
Ver 4.7 - App class is not visible in other classes of the project

### DISCUSSION THREAD

**Discussion by mcNets** [unoplatform/uno#11082] [Q&A] [answered]
I cannot access App class or any of its public properties from another class. (f.e. a Page or a ViewModel)

```
namespace MyNamespace
{
    public sealed partial class App : Application
    {
        public static Window MainWindow { get; private set; }
    }
}
```

If I try to access App class within MainWindow.xaml.cs I'm getting:

>The name App doesn't exists in the current context.

Even if I prefix it with the project namespace:

`MyNamespace.App`

Neither App.Current, that usually can be casted as ((App)App.Current)

**Comment by jeromelaban** [comment, marked as answer]
That is an expected behavior at this time. App.xaml needs to be placed in the head projects for both WinUI and Uno. Still, what you can do is make a base class for app.xaml.cs in the library (called `MyApp`), then inherit from that class in the real app.xaml.cs.

This is a valid concern for this template, and we may change the app.xaml to use that structure instead.

**Reply by mcNets** [reply]
In this way?

```
public class MyApp : Application { }

public sealed partial class App : MyApp
{
}
```

**Reply by jeromelaban** [reply]
Something like this, yes! You can basically move everything from the app.xaml.cs to that file, and you should get the same behavior.

**Reply by mcNets** [reply]
@jeromelaban I'm getting an error because `App.g.i.cs` declares the App class using a different base class. Maybe it's hardcoded in the source generator?

`partial class App : global::Microsoft.UI.Xaml.Application`

By now I'm using this approach:
```
public class MyApp
{
    public static Application? App { get; set; }
    public static Window? MainWindow { get; set; }
}
```

Assigning the properties in the App's constructor:
```
        public App()
        {
            MyApp.App = this;
            MyApp.MainWindow = MainWindow;
```
